packages/typespec-python/test/generic_mock_api_tests/test_payload_multipart.py

```diff
@@ -52,7 +52,7 @@ def client():
 def test_multi_part(client: MultiPartClient, op_name, model_class, data, file):
     op = getattr(client.form_data, op_name)
     # test bytes
-    body = {k: open(str(v), "rb").read() for k, v in file.items()}
+    body = {k: ("blob", open(str(v), "rb").read(), "application/octet-stream") for k, v in file.items()}
```

Comment:
Lack test for `binary_array_parts`/`complex`, and you can refer to my previous commit:
https://github.com/Azure/autorest.python/blob/fba4a91651b5cce6c8926b6170abd135e790a450/packages/typespec-python/test/azure/mock_api_tests/test_payload_multipart.py#L56-L71